File: python/problem_128_longest_consecutive_sequence.py
// https://leetcode.com/problems/longest-consecutive-sequence

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:

        num_set = set(nums)
        max_count = 0

        for num in num_set:
            if num-1 not in num_set:
                count=1
                while num+count in num_set:
                    count+=1
                max_count = max(count,max_count)
        
        return max_count

File: java/Problem128_LongestConsecutiveSequence.java
// https://leetcode.com/problems/longest-consecutive-sequence

class Solution {
    public int longestConsecutive(int[] nums) {
        Arrays.sort(nums);
        int count = 1;
        int max = 1;

        for (int i=1; i<nums.length; i++) {
            if (nums[i] == nums[i-1])
                continue;
                
            if (nums[i] == (nums[i-1]+1)) {
                count++;
                if (count > max) {
                    max = count;
                }
            } else {
                count = 1;
            }
        }

        return max;

    }
}
